src/detect-engine-event.c

```diff
@@ -160,6 +160,9 @@ static DetectEngineEventData *DetectEngineEventParse (const char *rawstr)
 
     if (de->event == STREAM_REASSEMBLY_OVERLAP_DIFFERENT_DATA) {
         StreamTcpReassembleConfigEnableOverlapCheck();
+    } else if (de->event == UDP_HLEN_INVALID) {
+        SCLogWarning("Rule uses decode-event \"udp.hlen_invalid\" which will be deprecated in "
```

Comment:
Just a note: Suricata 8.0 was not agreed upon, I just put it there as a safe milestone when to remove the keyword. So @victorjulien needs to confirm this. Alternatively, the keyword can be used for a case that Shivani fixed.